rocketpy/mathutils/function.py

```diff
@@ -396,14 +353,12 @@ def set_get_value_opt(self):
         x_data = self.x_array
         y_data = self.y_array
         x_min, x_max = self.x_initial, self.x_final
-        if self.__extrapolation__ == "zero":
-            extrapolation = 0  # Extrapolation is zero
-        elif self.__extrapolation__ == "natural":
-            extrapolation = 1  # Extrapolation is natural
-        elif self.__extrapolation__ == "constant":
-            extrapolation = 2  # Extrapolation is constant
-        else:
-            raise ValueError(f"Invalid extrapolation type {self.__extrapolation__}")
+        try:
+            extrapolation = self.EXTRAPOLATION_TYPES[self.__extrapolation__]
+        except KeyError as err:
+            raise ValueError(
+                f"Invalid extrapolation type {self.__extrapolation__}"
+            ) from err
```

Comment:
If there is validation for setting `self.__extrapolation__`  this try-except should never be needed